integrations/sonarqube/client.py

```diff
@@ -256,17 +292,27 @@ async def get_issues_by_component(
         """
         component_issues = []
         component_key = component.get("key")
+        query_params = {"componentKeys": component_key}
         logger.info(f"Fetching all issues in component: {component_key}")
+
+        selector = typing.cast(
+            SonarQubeIssueResourceConfig, event.resource_config
+        ).selector
+
+        if hasattr(selector, "api_query_params") and selector.api_query_params:
+            selector_query_params = selector.api_query_params.generate_request_params()
+            query_params.update(selector_query_params)
```

Comment:
could you make a function for this?
I see it's a duplicated in `get_components` and it will simplify the code